internal/bake/hcl/parser/schema.go

```diff
@@ -198,7 +198,21 @@ var BakeSchema = &schema.BodySchema{
 					},
 					"output": {
 						IsOptional: true,
-						Constraint: schema.List{Elem: schema.AnyExpression{OfType: cty.String}},
+						Constraint: schema.List{Elem: schema.OneOf{
+							schema.AnyExpression{OfType: cty.String},
+							schema.Object{
+								Attributes: map[string]*schema.AttributeSchema{
+									"type": {
+										IsOptional: true,
+										Constraint: schema.AnyExpression{OfType: cty.String},
+									},
+									"dest": {
+										IsOptional: true,
```

Comment:
```suggestion
										IsOptional: false,
```
```HCL
target "default" {
  output = [{ type = "cacheonly" }]
}
```
From local testing and the [example](https://docs.docker.com/build/bake/reference/#targetoutput) in the documentation, it seems like only `type` is required.